### PR TITLE
adding url session extensions

### PR DESCRIPTION
**Pull request by russellbstephens**

### CODE REVIEW

File: Snail/Extensions/URLResponseExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//  Copyright © 2017 Compass. All rights reserved.
+
+import Foundation
+
+extension URLResponse {
+    func isSuccessful() -> Bool {
```
Comment (wesbillman) [inline Snail/Extensions/URLResponseExtensions.swift]: `var isSuccess: Bool`?

File: Snail/Extensions/URLSessionExtension.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import UIKit
+
+extension URLSession {
+    enum ErrorType: Error {
+        case invalidData
+        case invalidResponse
+    }
+
+    public func observable(request: URLRequest) -> Observable<[String: AnyObject]> {
```
Comment (wesbillman) [inline Snail/Extensions/URLSessionExtension.swift]: Do we need `AnyObject` here? Can this be `Any` instead?

Comment (wesbillman) [inline Snail/Extensions/URLSessionExtension.swift]: Should we also have an `Observable<[Any]>` for handing json arrays?

File: Snail/Extensions/URLSessionExtension.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                return
+            }
+            observer.on(.next(dictionary))
+        }, onError: { observer.on(.error($0)) })
+        return observer
+    }
+
+    func data(request: URLRequest) -> Observable<(Data, URLResponse)> {
+        let observer = Replay<(Data, URLResponse)>(1)
+        let task = dataTask(with: request, completionHandler: { data, aResponse, error in
```
Comment (wesbillman) [inline Snail/Extensions/URLSessionExtension.swift]: can we rename `aResponse` to `response`?

File: Snail/Extensions/URLSessionExtension.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+import UIKit
+
+extension URLSession {
+    enum ErrorType: Error {
+        case invalidData
+        case invalidResponse
+    }
+
+    public func observable(request: URLRequest) -> Observable<[String: Any]> {
```
Comment (wesbillman) [inline Snail/Extensions/URLSessionExtension.swift]: this name seems weird `observable` that feels really confusing.

Why not follow the `data` pattern below?

Comment (wesbillman) [inline Snail/Extensions/URLSessionExtension.swift]:
```swift
public func dictionary(request: URLRequest)
//or
public func json(request: URLRequest)

//and for arrays...
public func array(request: URLRequest)
```

Comment (russellbstephens) [inline Snail/Extensions/URLSessionExtension.swift]: <3

**Pull request merged**